alephmarcreader/abstractalephmarcreader.py

```diff
@@ -353,4 +362,117 @@ def get_recipient_organisation(self):
+                remarks.append(footnote_text)
+
+        return remarks
+
+    get_supplement_remarks.__annotations__ = {'return': [str]}
+
+    def get_document_state(self):
+        """
+        Returns a description of the document state. (i.e. "original", "copy", "draft", etc.)
+        :return: [str]
+        """
+        states = []
+
+        for field in self.__get_field('250'):
+            footnote_text = self._handle_subfields_cardinality_max_one(self.__get_subfield_texts(field, 'a'))
+
+            if footnote_text:
+                states.append(footnote_text)
+
+        return states
+
+    get_document_state.__annotations__ = {'return': [str]}
+
+    def get_original_date_and_place(self):
+        """
+        Returns a transcription of the date and place, as given on the letter.
+        (If it starts with square bracket, it's conjectured.)
+        :return: [OriginalDate]
+        """
+        original_dates = []
+
+        for field in self.__get_field('264'):
+            date = self._handle_subfields_cardinality_max_one(self.__get_subfield_texts(field, 'c'))
+            place = self._handle_subfields_cardinality_max_one(self.__get_subfield_texts(field, 'a'))
+            # TODO Ignore Fields starting wit '['?
+
+            orig_date = self.OriginalDate(date, place)
+            original_dates.append(orig_date)
+
+        return original_dates
+
+    get_original_date_and_place.__annotations__ = {'return': [OriginalDate]}
+
+    def get_references_to_related_entries(self):
+        """
+        Returns a list of related entries.
+        That is usually the case, if there are several document stages of the same letter.
+        (E.g. Original, draft, copy, etc.)
+        :return: [str]
+        """
+        references = []
+
+        for field in self.__get_field('533'):
+            ref = self._handle_subfields_cardinality_max_one(self.__get_subfield_texts(field, 'a'))
+            if ref:
+                references.append(ref)
+
+            ref = self._handle_subfields_cardinality_max_one(self.__get_subfield_texts(field, 'n'))
+            if ref:
+                references.append(ref)
+
+        for field in self.__get_field('534'):
+            ref = self._handle_subfields_cardinality_max_one(self.__get_subfield_texts(field, 'n'))
+            if ref:
+                references.append(ref)
+
+        for field in self.__get_field('544'):
+            ref = self._handle_subfields_cardinality_max_one(self.__get_subfield_texts(field, 'n'))
+            if ref:
+                references.append(ref)
+
+        return references
+
+    get_references_to_related_entries.__annotations__ = {'return': [OriginalDate]}
+
+    def get_bibliographic_references(self):
+        """
+        Returns bibliographic references relevant to the letter. (Usually, where it has been printed.)
+        :return: [str]
+        """
+        references = []
+
+        for field in self.__get_field('581'):
+            ref = u''
```

Comment:
I don't understand what this prefix means.

Reply:
Subfield i contains information on what subfield a is.  
It can be things like "Druck" or "Inhaltsangabe bei" and stuff.  
a should be the the bibliographic reference itself (though that's not always the case, we also have cases, where i is missing and a is "Druck in: bla bla").

So basically, I consider a the body of the information, because tht's what counts. But in case we have Subfield i, I add this as a prefix to the core information, followed by a colon and a whitespace. (`[PREFIX: ]BODY` or `[SUBFIELD_I: ]SUBFIELD_A`,  
e.g. :000056870  
`Druck: Joh. I B. Briefe 1, p.444` for `<datafield tag="581" ind1=" " ind2=" ">
    <subfield code="i">Druck</subfield>
    <subfield code="a">Joh. I B. Briefe 1, p.444</subfield>
  </datafield>`)

To me, that seemed like a reasonable way to handle this, for now. And as soon as the propper bibliographies on BEOL are complete, this field should be obsolete anyway.  
But feel free to suggest a better solution.

Reply:
I was just thinking that if you need this information later, you would have to parse the string which would be cumbersome. You could return an instance of the class with separate members.

Reply:
ok, will do. In that case, I just add a method `get_pretty_string()` that returns them combined.

Reply:
is that better, now?

Reply:
That's nice. Could you test for the members `prefix` and `reference` in the test, too?

Reply:
voila :)